crates/worker/src/partition_processor_manager/spawn_processor_task.rs

```diff
@@ -117,15 +122,66 @@ impl SpawnPartitionProcessorTask {
             {
                 let options = options.clone();
                 let key_range = key_range.clone();
+
                 move || async move {
-                    let partition_store = partition_store_manager
-                        .open_partition_store(
-                            partition_id,
-                            key_range,
-                            OpenMode::CreateIfMissing,
-                            &options.storage.rocksdb,
-                        )
-                        .await?;
+                    let partition_store = if !partition_store_manager
+                        .has_partition_store(pp_builder.partition_id)
+                        .await
+                    {
+                        let snapshot = if snapshot_repository.is_none() {
+                            debug!(
+                                partition_id = %partition_id,
+                                "No snapshot repository configured",
+                            );
+                            None
+                        } else {
+                            debug!(
+                                partition_id = %partition_id,
+                                "Looking for partition snapshot from which to bootstrap partition store",
+                            );
+                            snapshot_repository.expect("is some").get_latest(partition_id).await?
+                        };
```

Comment:
nit: You could use `if let Some(snapshot_repository) = snapshot_respository { } else {}` instead of `is_none` and then `expecting`.

Reply:
I tried to get it to work this way initially but the code ended up being deeply nested and looked less readable to my eyes. I'll give this another pass when I come back to this.